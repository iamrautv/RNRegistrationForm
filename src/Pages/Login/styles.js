import { StyleSheet } from 'react-native';
import { theme } from '../../config';

const { colors } = theme;

const styles = StyleSheet.create({
  wrap: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    backgroundColor: colors.white,
    alignItems: 'center'
  },
  loginLogoWrap: {
    width: '100%',
    height: 500,
    justifyContent: 'center',
    alignItems: 'center',
  },
  loginLogoImg: {
    width: 250,
  },
  loginBtn: {
    backgroundColor: colors.primary,
    paddingVertical: 15,
    paddingHorizontal: 35,
    borderRadius: 15
  },
  loginBtnTxt: {
    color: colors.white,
    fontWeight: '700',
    fontSize: 19
  },
  loginLinkWrap: {
    flexDirection: 'row',
    marginTop: 15
  },
  loginLinkPlainTxt: {
    fontSize: 16,
    marginRight: 5
  },
  loginLinkTxt: {
    color: colors.link,
    fontSize: 16
  }
});

export default styles;
