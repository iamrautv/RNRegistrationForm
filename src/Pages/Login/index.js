import React from 'react';
import {
  View,
  Text,
  Image,
  Button,
  TouchableOpacity
} from 'react-native';
import { images, constants } from '../../config';
import styles from './styles';

const { routes } = constants;

const Login = (props) => {
  const { navigation } = props;

  return (
    <View style={styles.wrap}>
      <View style={styles.loginLogoWrap}>
        <Image
          source={images.loginLogo}
          resizeMode="contain"
          style={styles.loginLogoImg}
        />
      </View>
      <TouchableOpacity
        onPress={() => { navigation.navigate(routes.register) }}
        style={styles.loginBtn}
      >
        <Text style={styles.loginBtnTxt}>Join Our School</Text>
      </TouchableOpacity>
      <View style={styles.loginLinkWrap}>
        <Text style={styles.loginLinkPlainTxt}>Existing User?</Text>
        <TouchableOpacity
          onPress={() => { navigation.navigate(routes.register) }}
        >
          <Text style={styles.loginLinkTxt}>Login</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default Login;
