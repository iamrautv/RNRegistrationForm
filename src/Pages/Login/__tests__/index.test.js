/**
 * @format
 */

import 'react-native';
import React from 'react';
import Login from '../index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Renders Login page', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
