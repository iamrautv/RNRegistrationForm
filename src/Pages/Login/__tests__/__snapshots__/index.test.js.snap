// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Login page 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 500,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/LoginLogo.png",
        }
      }
      style={
        Object {
          "width": 250,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#008080",
        "borderRadius": 15,
        "opacity": 1,
        "paddingHorizontal": 35,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 19,
          "fontWeight": "700",
        }
      }
    >
      Join Our School
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginRight": 5,
        }
      }
    >
      Existing User?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1f98d7",
            "fontSize": 16,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
